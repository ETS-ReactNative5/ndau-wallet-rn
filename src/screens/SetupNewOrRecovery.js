import React, { Component } from 'react'
import {
  View,
  ScrollView,
  Text,
  Platform,
  TouchableWithoutFeedback,
  BackHandler,
  StatusBar
} from 'react-native'
import CommonButton from '../components/CommonButton'
import cssStyles from '../css/styles'
import SetupStore from '../model/SetupStore'
import { SafeAreaView } from 'react-navigation'
import SetupProgressBar, {
  NEW_WALLET_SETUP_TYPE,
  RECOVERY_WALLET_SETUP_TYPE
} from '../components/SetupProgressBar'
import SetupGetRecoveryPhrase from '../screens/SetupGetRecoveryPhrase'
import AppConstants from '../AppConstants'

class SetupNewOrRecovery extends Component {
  constructor (props) {
    super(props)

    this.state = {
      toggleCount: 1,
      maxToggle: 10
    }
  }

  componentWillUnmount () {
    BackHandler.removeEventListener('hardwareBackPress', this.handleBackButton)
  }

  handleBackButton () {
    return true
  }

  componentDidMount () {
    BackHandler.addEventListener('hardwareBackPress', this.handleBackButton)
  }

  showNewWallet = () => {
    this.props.navigation.navigate('SetupYourWallet', {
      walletSetupType: 'new'
    })
  }

  showUseExistingRecovery = () => {
    this.props.navigation.navigate('SetupGetRecoveryPhrase', {
      walletSetupType: 'recovery',
      mode: AppConstants.NORMAL_MODE,
      walletSetupType: RECOVERY_WALLET_SETUP_TYPE
    })
  }

  render () {
    return (
      <SafeAreaView style={cssStyles.safeContainer}>
        <StatusBar barStyle='light-content' backgroundColor='#1c2227' />
        <View style={cssStyles.container}>
          <ScrollView style={cssStyles.contentContainer}>
            <SetupProgressBar navigation={this.props.navigation} />
            <View>
              <Text style={cssStyles.wizardText}>
                Welcome to ndau, a cryptocurrency designed to be a buoyant long-term store of value.
                {' '}
                {Platform.OS === 'android' ? '\n' : ''}
              </Text>
            </View>
            <View>
              <Text style={cssStyles.wizardText}>
                To get started securely, you will create a new wallet, protect it with a password,
                and create a recovery phrase which you will need in order to restore your wallet if
                you lose access to it.
                {Platform.OS === 'android' ? '\n' : ''}
              </Text>
            </View>
          </ScrollView>
          <View style={cssStyles.footer}>
            <View style={cssStyles.navButtonWrapper}>
              <CommonButton
                onPress={() => this.showUseExistingRecovery()}
                title='Recover a wallet (less common)'
              />
            </View>
            <View style={cssStyles.navButtonWrapper}>
              <CommonButton
                onPress={() => this.showNewWallet()}
                title='New wallet'
              />
            </View>
          </View>
        </View>
      </SafeAreaView>
    )
  }
}

export default SetupNewOrRecovery
