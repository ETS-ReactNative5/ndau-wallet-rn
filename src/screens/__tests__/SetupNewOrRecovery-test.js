import { mount } from 'enzyme'
import React from 'react'
import { StyleSheet } from 'react-native'
import SetupNewOrRecovery from '../SetupNewOrRecovery'

import renderer from 'react-test-renderer'

describe('testing SetupNewOrRecovery...', () => {
  let styles = StyleSheet.create({
    wizardText: {
      color: '#ffffff',
      fontSize: 20
    }
  })
  const navigator = {
    setStyle: () => {},
    toggleNavBar: () => {}
  }

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <SetupNewOrRecovery parentStyles={styles} navigator={navigator} />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('can click the button', () => {
    const wrapper = mount(
      <SetupNewOrRecovery navigator={navigator} parentStyles={styles} />
    )
    const onlyButton = wrapper.find('Button').at(0)
    onlyButton.simulate('click')
  })
})
