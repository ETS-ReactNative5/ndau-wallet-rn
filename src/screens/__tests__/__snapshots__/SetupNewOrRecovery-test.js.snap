// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing SetupNewOrRecovery... renders correctly 1`] = `
<View
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "backgroundColor": "#1c2227",
      "flex": 1,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#1c2227",
        "flex": 1,
        "paddingBottom": 26.5,
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <View>
        <Image
          resizeMode="contain"
          screenNumber={1}
          source={
            Object {
              "testUri": "../../../img/step-1.png",
            }
          }
          style={
            Object {
              "backgroundColor": "transparent",
              "height": 60,
              "marginBottom": 5,
              "width": "100%",
            }
          }
        />
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#ffffff",
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 20,
              }
            }
          >
            Welcome to ndau, a cryptocurrency designed to be a buoyant long-term store of value.
             
            
          </Text>
        </View>
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#ffffff",
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 20,
              }
            }
          >
            To get started securely, you will create a new wallet, protect it with a password, and create a recovery phrase which you will need in order to restore your wallet if you lose access to it.
            
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        style={
          Object {
            "justifyContent": "space-between",
            "marginTop": 26.5,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityTraits="button"
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignContent": "center",
              "backgroundColor": "#4e957a",
              "borderRadius": 3,
              "height": 93.5,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#ffffff",
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Recover a wallet (less common)
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "justifyContent": "space-between",
            "marginTop": 26.5,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityTraits="button"
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignContent": "center",
              "backgroundColor": "#4e957a",
              "borderRadius": 3,
              "height": 93.5,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#ffffff",
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            New wallet
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
